src/app/JoinServer.js

```diff
@@ -46,20 +43,20 @@ class JoinServer {
 			name: 'main',
 			level: logLevel,
 		}),
-		dataUnionClient = new DataUnionClient({
-			auth: {
-				privateKey,
-			}
-		}),
-		signedRequestValidator = SignedRequestValidator.validator,
-		joinRequestService = new service.JoinRequestService(logger, dataUnionClient, onMemberJoin),
+		signedRequestValidator = rest.SignedRequestValidator.validator,
+		joinRequestService = undefined,
 	} = {}) {
 
 		this.expressApp = expressApp
 		this.logger = logger
-		this.dataUnionClient = dataUnionClient
 		this.signedRequestValidator = signedRequestValidator
 		this.customJoinRequestValidator = customJoinRequestValidator
+		if (!joinRequestService) {
```

Comment:
To decrease manual maintenance work when new chains are added, could this part read the available chains from the config package? It could filter from the config all the chain configs that define a `DataUnionFactory`.